src/adapters/aave/AaveAdapter.sol

```diff
@@ -2,32 +2,77 @@
 pragma solidity 0.8.10;
 
 // External references
+import {IERC20Metadata} from "@openzeppelin/contracts/token/ERC20/extensions/IERC20Metadata.sol";
 import {SafeERC20, IERC20} from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
-import "../../interfaces/aave/ILendingPool.sol";
+import {IAToken} from "../../interfaces/aave/IAToken.sol";
+import {ILendingPool} from "../../interfaces/aave/ILendingPool.sol";
 
 // Internal references
 import "../BaseAdapter.sol";
 
-/// @title AaveV3 Adapter
-/// NOTE: Aave V3 https://docs.aave.com/developers/tokens/atoken
+/// @title AaveV2 Adapter
+/// NOTE: Aave V2 https://docs.aave.com/developers/tokens/atoken
 contract AaveAdapter is BaseAdapter {
     using FixedMath for uint256;
-    using SafeERC20 for IERC20;
+    using SafeERC20 for IERC20Metadata;
 
-    constructor(AdapterParams memory _adapterParams) BaseAdapter(_adapterParams) {}
+    // TODO: Select LendingPool by ChainId
+    address public constant LENDING_POOL_V2_MAINNET = 0x7d2768dE32b0b80b7a3454c06BdAc94A69DDc7A9;
+
+    ILendingPool internal _pool;
+
+    constructor(AdapterParams memory _adapterParams) BaseAdapter(_adapterParams) {
+        require(adapterParams.underlying == underlying(), "AaveAdapter: unmatching underlying");
```

Comment:
Added this to check the validity of adapter params. Not sure if this is necessary